sample-app/src/components/DropdownSection.tsx

```diff
@@ -0,0 +1,108 @@
+import classNames from "classnames";
+import { useState, useEffect } from "react";
+
+export interface Option {
+  value: string,
+  display: JSX.Element
+}
+
+interface DropdownSectionProps<T> {
+  focusStatus?: string,
+  options: T[],
+  optionIdPrefix: string,
+  onFocusChange: (value: string, focusedOptionId: string) => void,
+  onLeaveSectionFocus?: (focusNext: boolean) => void,
+  onClickOption: (option: T, optionIndex: number) => void,
+  label?: string,
+  cssClasses: {
+    sectionContainer: string,
+    sectionLabel: string,
+    optionsContainer: string,
+    option: string,
+    focusedOption: string
+  }
+}
+
+export default function DropdownSection<T extends Option>({
+  focusStatus,
+  options,
+  optionIdPrefix,
+  onFocusChange,
+  onLeaveSectionFocus = () => {},
+  onClickOption,
+  label = '',
+  cssClasses
+}: DropdownSectionProps<T>): JSX.Element | null {
+
+  const [focusedOptionIndex, setFocusedOptionIndex] = useState<number>(0);
+
+  function changeOptionFocus(focusNext: boolean) {
+    let newIndex = focusNext ? focusedOptionIndex + 1 : focusedOptionIndex - 1;
+
+    if (newIndex > -1 && newIndex < options.length) {
+      onFocusChange(options[newIndex].value, `${optionIdPrefix}-${newIndex}`);
+    } else {
+      focusNext ? onLeaveSectionFocus(true) : onLeaveSectionFocus(false);
+      newIndex = newIndex === -1 ? 0 : options.length - 1;
+    }
+
+    setFocusedOptionIndex(newIndex);
+  }
+
+  function handleKeyDown(evt: globalThis.KeyboardEvent) {
+    if (focusStatus === 'active') {
+      if (['ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(evt.key)) {
+        evt.preventDefault();
+      }
+
+      if (evt.key === 'ArrowDown' || evt.key === 'ArrowRight') {
+        changeOptionFocus(true);
+      } else if (evt.key === 'ArrowUp' || evt.key === 'ArrowLeft') {
+        changeOptionFocus(false);
+      }
+    }
+  }
+
+  useEffect(() => {
+    if (focusStatus === 'active') {
+      onFocusChange(options[focusedOptionIndex].value, `${optionIdPrefix}-${focusedOptionIndex}`);
+    } else if (focusStatus === 'reset') {
+      setFocusedOptionIndex(0);
+    }
+  }, [focusStatus]);
+
+  useEffect(() => {
```

Comment:
can this useEffect be group with the useEffect above?

Reply:
the event listener needs to be added whenever the section is active, not just when it first becomes active (like the first useEffect). I tried combining them, but then only the first keydown event is registered

Reply:
you are cloning the element each time when section is set active right? So wouldn't it be first render of the newly clone section be enough to setup the listener?